### ISSUE TITLE
Pretrained model does not match the model definition

### ISSUE THREAD

**Issue by nisha1729** [jellycsc/building-autoencoders-in-Pytorch#1]
The pre-trained model weights could not be loaded to the current model definition. I changed the model to 
```
self.encoder = nn.Sequential(
            nn.Conv2d(3, 32, 4, stride=2, padding=1),            # [batch, 12, 16, 16]
            nn.ReLU(),
            nn.Conv2d(32, 48, 3, stride=2, padding=1),           # [batch, 24, 8, 8]
            nn.ReLU(),
        )
        self.decoder = nn.Sequential(
		    nn.ConvTranspose2d(48, 32, 3, stride=2, padding=1),  # [batch, 24, 8, 8]
            nn.ReLU(),
			nn.ConvTranspose2d(32, 3, 4, stride=2, padding=1),   # [batch, 12, 16, 16]
            nn.ReLU(),
        )
```
Now the code works, but the reconstruction is quite bad.
![image](https://user-images.githubusercontent.com/7227383/126974855-06417241-3094-464e-bea7-adae3be513b5.png)

Do you have a different pre-trained model that can be used? Thanks!

**Comment by nisha1729** [comment]
Also, the loss should be MSELoss() instead of BCELoss().

**Comment by ColdAsYou165** [comment]
I have the same question too!It seems that the author didn't provides the right weights?

**Comment by ColdAsYou165** [comment]
Have you solved the problem yet

**Comment by nisha1729** [comment]
I just retrained the model 😅

**Comment by Wsine** [comment]
I got the same problem

**Comment by JerrySongCST** [comment]
> The pre-trained model weights could not be loaded to the current model definition. I changed the model to
> 
> ```
> self.encoder = nn.Sequential(
>             nn.Conv2d(3, 32, 4, stride=2, padding=1),            # [batch, 12, 16, 16]
>             nn.ReLU(),
>             nn.Conv2d(32, 48, 3, stride=2, padding=1),           # [batch, 24, 8, 8]
>             nn.ReLU(),
>         )
>         self.decoder = nn.Sequential(
> 		    nn.ConvTranspose2d(48, 32, 3, stride=2, padding=1),  # [batch, 24, 8, 8]
>             nn.ReLU(),
> 			nn.ConvTranspose2d(32, 3, 4, stride=2, padding=1),   # [batch, 12, 16, 16]
>             nn.ReLU(),
>         )
> ```
> 
> Now the code works, but the reconstruction is quite bad. ![image](https://user-images.githubusercontent.com/7227383/126974855-06417241-3094-464e-bea7-adae3be513b5.png)
> 
> Do you have a different pre-trained model that can be used? Thanks!

Yes, the network that provided by the author doesn't match with the provided weights. Your modification were almost correct, if you change the last row " nn.ReLU()," to " nn.Sigmoid(),", you will get the correct predictions!
